standalone-metastore/metastore-common/src/main/java/org/apache/hadoop/hive/metastore/Warehouse.java

```diff
@@ -448,6 +448,20 @@ public boolean renameDir(Path sourcePath, Path destPath, boolean needCmRecycle)
     return false;
   }
 
+  public boolean copyDir(Path sourcePath, Path destPath, boolean needCmRecycle) throws MetaException {
+    try {
+      if (needCmRecycle) {
+        cm.recycle(sourcePath, RecycleType.COPY, true);
+      }
+      FileSystem srcFs = getFs(sourcePath);
+      FileSystem destFs = getFs(destPath);
+      return FileUtils.copy(srcFs, sourcePath, destFs, destPath, false, false, conf);
```

Comment:
minor comment. This can turn out to be expensive depending on size of the directory, size and number of files. May be good to add a TODO for later ref/fix.

Reply:
done